boards/shared/Inc/App/App_SharedSignal.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#include <stdbool.h>
+#include <stdint.h>
+#include "configs/App_SharedSignalConfig.h"
+
+#ifndef World
+#error "Please define the 'World' type"
+#endif
+
+struct Signal;
+
+struct SignalCallback
+{
+    // How long the signal must be continuously high for, in milliseconds,
+    // before the callback function is triggered
+    uint32_t high_duration_ms;
+
+    // The callback function
+    void (*function)(struct World *);
+};
+
+/**
+ * Allocate and initialize a signal
+ * @param initial_time_ms The initial time, in milliseconds, used to initialize
+ *                        the internal state of the signal
+ * @param is_high A function that can be called to check if the signal is high
+ * @param callback The signal callback for the signal
+ * @return The created signal, whose ownership is given to the caller
+ */
+struct Signal *App_SharedSignal_Create(
+    uint32_t initial_time_ms,
+    bool (*is_high)(struct World *),
+    struct World *        world,
```

Comment:
Since `world` is used by the `callback` _and_ `is_high`, I thought it would make sense to put it in `struct Signal` as opposed to `struct SignalCallback`.